File: src/aocpython/day02/checksum.py
from itertools import permutations

def part1():
    return sum(max(row) - min(row) for row in data)

def part2():
    return sum([[x//y for x, y in permutations(row, 2) if x % y == 0][0] for row in data])

with open('./data/02') as puzzle_input:
    data = [[int(d) for d in row.split('\t')] for row in puzzle_input]
    print(part1(), part2())

File: src/AocHaskell/Day02/Checksum.hs
import System.IO (readFile)
import Data.List.Split (splitOn)

readSpreadsheet :: String -> [[Int]]
readSpreadsheet = map (map read . (splitOn "\t")) . lines

maxMinChecksum :: [[Int]] -> Int
maxMinChecksum = sum . map (\row -> (maximum row) - (minimum row))

evenlyDivisibleChecksum :: [[Int]] -> Int
evenlyDivisibleChecksum = sum . map (\row -> head [x `div` y | x <- row, y <- row, x /= y, x `mod` y == 0]) 

main :: IO()
main = do   puzzleInput <- readFile "./data/02"
            print . maxMinChecksum $ readSpreadsheet puzzleInput
            print . evenlyDivisibleChecksum $ readSpreadsheet puzzleInput